substrate/frame/recovery/src/lib.rs

```diff
@@ -236,6 +239,9 @@ pub mod pallet {
 			+ GetDispatchInfo
 			+ From<frame_system::Call<Self>>;
 
+		/// Provider for the block number. Normally this is the `frame_system` pallet.
+		type BlockNumberProvider: BlockNumberProvider<BlockNumber = BlockNumberFor<Self>>;
```

Comment:
We shouldn't force the type to be the same as chains block number. see https://github.com/paritytech/polkadot-sdk/pull/5723
```suggestion
		type BlockNumberProvider: BlockNumberProvider;
```

Reply:
Yep, I have already noticed that after reviewing some PRs. That's why I mark this as draft.